File: 프로그래머스/1/12918. 문자열 다루기 기본/문자열 다루기 기본.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool solution(string s) {
    bool answer = false;
    bool isLength =false;
    bool isNum = false;
    
    if(all_of(s.begin(), s.end(), ::isdigit)){
        isNum = true;
    }
    
    if(s.length()==4 || s.length()==6 ){
        isLength = true;
    }
    
    if(isNum==true && isLength == true){
        answer = true;
    }
    
    
    
    return answer;
}

File: 프로그래머스/1/12918. 문자열 다루기 기본/문자열 다루기 기본.java
class Solution {
    public boolean solution(String s) {
        
        boolean answer = false;
        
        if( (s.length() == 4 || s.length() == 6) ){
            answer = true;
            for(char ch: s.toCharArray()){
                if(!Character.isDigit(ch)){
                    answer = false;
                    break;
                }
            }
            
        } 
        
        return answer;
        
        
        
        
        
        
//         boolean answer = false;
        
        
//         if( (s.length()==4 || (s.length()==6)) ){
            
//             answer = true;
            
//             for(int i=0; i< s.length(); i++){
                
//                 if(!Character.isDigit(s.charAt(i))){
//                     answer = false;
//                     break;
                    
//                 }
//             }
//         }
        
        
//         return answer;
    }
}